lib/bolt/cli.rb

```diff
@@ -551,6 +557,114 @@ def list_groups
       outputter.print_groups(groups)
     end
 
+    def new_plan(plan_name)
+      @logger.warn("Command 'bolt plan new' is experimental and subject to breaking changes.")
+
+      if config.project.name.nil?
+        raise Bolt::Error.new(
+          "Project directory '#{config.project.path}' is not a named project. Unable to create "\
+          "a project-level plan. To name a project, set the 'name' key in a 'bolt-project.yaml' "\
+          "configuration file.",
+          "bolt/unnamed-project-error"
+        )
+      end
+
+      if plan_name !~ Bolt::Project::NAME_REGEX
+        message = <<~MESSAGE.chomp
+          Invalid plan name '#{plan_name}'. Plan names are composed of one or more name segments
+          separated by double colons '::'.
+          
+          Each name segment must begin with a lowercase letter, and may only include lowercase
+          letters, digits, and underscores.
+          
+          Examples of valid plan names:
+              - myproject
+              - my_project::my_plan
+
+          Examples of invalid plan names:
+              - MyProject
+              - my-project::my-plan
+        MESSAGE
+
+        raise Bolt::ValidationError, message
+      end
+
+      name_segments = plan_name.split('::')
```

Comment:
You might be able to do `prefix, *segments, basename = plan_name.split('::')` and then `unless prefix == config.project.name`, `basename ||= 'init'` etc. Not _totally_ sure about all cases though.

Reply:
This is what I was originally trying to do but couldn't get it to work correctly. Didn't know you needed the splat to get an array of leftovers. :P